src/main/java/com/redhat/rhjmc/containerjfr/platform/ServiceRef.java

```diff
@@ -41,36 +41,37 @@
  */
 package com.redhat.rhjmc.containerjfr.platform;
 
+import javax.management.remote.JMXServiceURL;
+
 import org.apache.commons.lang.builder.EqualsBuilder;
 import org.apache.commons.lang.builder.HashCodeBuilder;
 import org.apache.commons.lang3.builder.ToStringBuilder;
 
 public class ServiceRef {
 
-    private final String connectUrl;
+    private final JMXServiceURL JMXServiceURL;
     private final String alias;
-    private final int port;
 
-    public ServiceRef(String connectUrl, int port) {
-        this(connectUrl, connectUrl, port);
+    public ServiceRef(JMXServiceURL JMXServiceURL) {
+        this(JMXServiceURL, JMXServiceURL.toString());
     }
 
-    public ServiceRef(String connectUrl, String alias, int port) {
-        this.connectUrl = connectUrl;
+    public ServiceRef(JMXServiceURL JMXServiceURL, String alias) {
+        this.JMXServiceURL = JMXServiceURL;
         this.alias = alias;
-        this.port = port;
     }
 
-    public String getConnectUrl() {
-        return connectUrl;
+    public ServiceRef(String host, String alias, int port) throws Exception {
```

Comment:
`host`/`alias`/`port` feels like an unnatural ordering of these params IMO. Maybe host/port/alias? Or alias/host/port? I think the host/port should be adjacent to each other.

Reply:
fixed.